### DISCUSSION TITLE
Rate limiting - sliding or fixed window, or something else?

### DISCUSSION THREAD

**Discussion by ogp-weeloong** [taskforcesh/bullmq#2524] [General]
Hi all,

I was wondering if BullMQ used a sliding or fixed window for rate limited queues? (or perhaps something else like token bucket?)

Thank you very much!

**Comment by manast** [comment]
Fixed window where the window is user-defined, here is everything you ever wanted to know about BullMQ rate-limiting: https://blog.taskforce.sh/rate-limit-recipes-in-nodejs-using-bullmq/

**Comment by ogp-weeloong** [comment]
Thank you!
